### ISSUE TITLE
Updating `DisplayBuild` to handle new winit `EventsLoop` API

### ISSUE THREAD

**Issue by mitchmindtree** [tomaka/glium#1579]
I've been making a start on updating glium as a follow up to the changes introduced in tomaka/glutin#864 in the hope that it lands.

The first issue I have ran into is updating the `glutin_backend` code. In the implementation of `DisplayBuild::build_glium_debug` for `glutin::WindowBuilder`, we require calling the `build` method on the `glutin::WindowBuilder`. This method requires access to a `&glutin::EventsLoop` which is currently not accessible to us within the impl of the `DisplayBuild::build_glium_debug` method.

I think the easiest way to solve this might be to add an associated `type EventsLoop` to the `DisplayBuild` trait and then change each of the build method signatures to take a `&Self::EventsLoop` as an argument. What are your thoughts on this? Let me know if you'd be happy with this or if you would prefer I take another approach.

**Comment by tomaka** [comment]
I don't think the `DisplayBuild` trait should be modified. Either kept as is or removed, but not modified.
After all other backends don't require an events loop.

**Comment by mitchmindtree** [comment]
You're right, adding an associated `EventsLoop` type would be quite a hackish glutin-specific fix.

The more I search through the code, the less I'm sure about the purpose of the `DisplayBuild` trait? It doesn't seem like any glium code requires being generic over `DisplayBuild` types, and I can't seem to find any implementations of `DisplayBuild` for any other backends in the wild. I came across this glium-sdl2 crate, however they [seem to implement their own `DisplayBuild` trait](https://github.com/nukep/glium-sdl2/blob/master/src/lib.rs#L157), and don't seem to be using the `glium`-provided one whatsoever.

Is the `DisplayBuild` trait purely for encouraging a cohesive API across different backends? If so, perhaps we could remove it in favour of something glutin-specific in the `glutin_backend` module?

**Comment by mitchmindtree** [comment]
Currently the API for building a glium display is as follows:

```rust
use glium::DisplayBuild;
let display = glium::glutin::WindowBuilder::new()
    .build_glium()
    .unwrap();
```

I wonder if we could change the API to look more like this:

```rust
let events_loop = glium::glutin::EventsLoop::new();
let window = glium::glutin::WindowBuilder::new()
    .build(&events_loop)
    .unwrap();
let display = glium::build(window).unwrap();
```

This way `glium` does not try to wrap the "builder" pattern of the window it is wrapping, but still takes ownership of the window. This also means the user doesn't require importing the `DisplayBuild` trait. We could still keep the `DisplayBuild` trait, but use it as a bound on the argument to the `glium::build` function I use in the example above. In this case, `DisplayBuild` would be implemented for `glutin::Window` rather than `glutin::WindowBuilder`. Thoughts?

**Comment by tomaka** [comment]
To be fair I don't remember why I went for `DisplayBuild`.
I guess building the glium display separately is the best solution.
